clientlib/build.gradle

```diff
@@ -0,0 +1,72 @@
+plugins {
+    // Provide convenience executables for trying out the examples.
+    id 'application'
+    // ASSUMES GRADLE 2.12 OR HIGHER. Use plugin version 0.7.5 with earlier gradle versions
+    id 'com.google.protobuf' version '0.8.8'
+    // Generate IntelliJ IDEA's .idea & .iml project files
+    id 'idea'
+    // Publish clientlib to maven central
+    id 'maven-publish'
+}
+
+repositories {
+    maven { // The google mirror is less flaky than mavenCentral()
+        url "https://maven-central.storage-download.googleapis.com/repos/central/data/"
+    }
+    mavenLocal()
+}
+
+sourceCompatibility = 1.12
+targetCompatibility = 1.12
+startScripts.enabled = false
+
+def grpcVersion = '1.26.0' // CURRENT_GRPC_VERSION
+def protobufVersion = '3.11.1'
+def protocVersion = protobufVersion
+
+dependencies {
+    //grpc deps
+    compile "io.grpc:grpc-protobuf:${grpcVersion}"
+    implementation "io.grpc:grpc-stub:${grpcVersion}"
+    implementation "io.grpc:grpc-okhttp:${grpcVersion}"
+    implementation "javax.annotation:javax.annotation-api:1.2"
+
+    // examples/advanced need this for JsonFormat
+    compile "com.google.protobuf:protobuf-java-util:${protobufVersion}"
+
+    runtimeOnly "io.grpc:grpc-netty-shaded:${grpcVersion}"
+
+    //test deps
+    compile "io.grpc:grpc-testing:${grpcVersion}"
+}
+
+sourceSets {
+    all {
+        proto {
+            srcDirs 'proto'
+        }
+        java {
+            srcDirs 'java'
+        }
+    }
+}
+
+protobuf {
+    protoc { artifact = "com.google.protobuf:protoc:${protocVersion}" }
+    plugins {
+        grpc { artifact = "io.grpc:protoc-gen-grpc-java:${grpcVersion}" }
+    }
+    generateProtoTasks {
+        all()*.plugins { grpc {} }
+    }
+}
+
+publishing {
+    publications {
+        maven(MavenPublication) {
+            groupId = 'org.apache.platypus'
```

Comment:
@umeshdangat are we going to change the name of the repo ? If yes, then we probably shouldn't publish to the public maven repo with this name....although I guess there's no harm even if we do?

Reply:
I also would like to confirm that it's fine publishing with org.apache in the package when this is not an official apache project but uses apache license.

Reply:
Given: https://issues.sonatype.org/browse/OSSRH-56201, @alok-sm and I decided we will simply publish this jar on internal nexus server for now.